opensrp-app/src/main/java/org/smartregister/util/SyncUtils.java

```diff
@@ -62,7 +63,9 @@ public void logoutUser() throws AuthenticatorException, OperationCanceledExcepti
 
     public void logoutUser(@StringRes int logoutMessage) throws AuthenticatorException, OperationCanceledException, IOException {
         //force remote login
-        opensrpContext.userService().forceRemoteLogin(opensrpContext.allSharedPreferences().fetchRegisteredANM());
+        if (!CoreLibrary.getInstance().isAllowOfflineLoginWithInvalidToken() || (HttpStatus.SC_UNAUTHORIZED != CoreLibrary.getInstance().context().allSharedPreferences().getLastAuthenticationHttpStatus())) {
+            opensrpContext.userService().forceRemoteLogin(opensrpContext.allSharedPreferences().fetchRegisteredANM());
+        }
```

Comment:
Consider accessing CoreLibrary instance through the field already defined in the class